src/interactive/repl.ts

```diff
@@ -540,22 +546,46 @@ export async function replStartDebugger(pipename: string) {
 export function activate(context: vscode.ExtensionContext) {
     g_context = context
 
+    let evaluationIndicator: vscode.StatusBarItem
+
     context.subscriptions.push(
         // listeners
         onSetLanguageClient(languageClient => {
             g_languageClient = languageClient
         }),
         onInit(connection => {
+            evaluationIndicator = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left)
+            evaluationIndicator.text = '$(refresh~spin) Julia: Evaluating ...'
+
             connection.onNotification(notifyTypeDisplay, plots.displayPlot)
             connection.onNotification(notifyTypeDebuggerRun, debuggerRun)
             connection.onNotification(notifyTypeDebuggerEnter, debuggerEnter)
-            connection.onNotification(notifyTypeReplStartEval, () => g_onStartEval.fire(null))
-            connection.onNotification(notifyTypeReplFinishEval, () => g_onFinishEval.fire(null))
+            connection.onNotification(notifyTypeReplStartEval, info => g_onStartEval.fire(info))
+            connection.onNotification(notifyTypeReplFinishEval, () => g_onFinishEval.fire(undefined))
             connection.onNotification(notifyTypeShowProfilerResult, showProfileResult)
             connection.onNotification(notifyTypeShowProfilerResultFile, showProfileResultFile)
-            connection.onNotification(notifyTypeProgress, updateProgress)
+            connection.onNotification(notifyTypeProgress, progress => {
+                evaluationIndicator.hide() // otherwise evaluation indicator and progress indicator will look verbose
+                updateProgress(progress)
+            })
+        }),
+        onStartEval(info => {
+            evaluationIndicator.tooltip = `click to jump to the code`
+            if (info) {
+                evaluationIndicator.command = Command.create('openFile', 'language-julia.openFile', {
+                    path: info.filename,
+                    line: info.line + 1,
+                })
+            }
+            evaluationIndicator.show()
```

Comment:
it might be more useful if click on the indicator interrupts the execution ?

Reply:
Yeah. What do you think about https://github.com/julia-vscode/julia-vscode/pull/1735 as an alternative to this?